### PR TITLE
adding lesson 6 assignment

### PR DESCRIPTION
**Pull request by jayjonesseattle**
Please take a look at mailroom part 4 in lesson 6. Are there other parts of the code that you suggest I write a test for? As always, I welcome your feedback.

### CODE REVIEW

File: students/terrance_jones/lesson6/test_mailroom_part4.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+def test_add_or_update_donor():
+    mailroom_part4.add_or_update_donor('Carlos Santos', 10) 
+    assert donor_list['Carlos Santos'] == [25,50,100,10]
+
+    mailroom_part4.add_or_update_donor('New Donor', 100)
+    assert donor_list['New Donor']==[100]
+
+def test_thank_you_letter():
+    assert thank_you_letter('Carlos Santos', 500) == "Dear Carlos Santos, \n Thank you for your $500.00 donation. \nSincerely, Mailroom."
```
Comment (natasha-aleksandrova) [inline students/terrance_jones/lesson6/test_mailroom_part4.py]: this is a really good start for tests.

couple other tests you may write:
* report generation
* file generation 

please review guidelines here for more information.
https://uwpce-pythoncert.github.io/PythonCertDevel/exercises/mailroom-part4.html#create-report
https://uwpce-pythoncert.github.io/PythonCertDevel/exercises/mailroom-part4.html#send-letters

if you still have questions, I'll be happy to extend on it.

File: students/terrance_jones/lesson6/test_mailroom_part4.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+
+import mailroom_part4
+from mailroom_part4 import donor_list
+from mailroom_part4 import donor_exists
+from mailroom_part4 import add_or_update_donor
+from mailroom_part4 import thank_you_letter
```
Comment (natasha-aleksandrova) [inline students/terrance_jones/lesson6/test_mailroom_part4.py]: note that you can also do a one liner import on above:
`from mailroom_part4 import donor_list, donor_exists  # etc`

**Comment by jayjonesseattle** [comment]
Hi Natasha, I'm having some trouble getting lesson 7 assignment started. I
feel like I understood the content but I don't think that I am
understanding the instructions for it exactly. Maybe show some things are
worded are confusing.  This is pretty much what I have so far but honestly,
as I read through the assignment instructions I get confused. Any advice?


class Element(object):
    """renders an html element"""
    tag_name = ""
    indent = "     "

    def __init__(self, content=None):
        if content is None:
            self.content = []
        else:
            self.content = [content]


    def append_content(self, new_content):
        self.content.append(new_content)

    def render(self, file_out, cur_ind = ""):
        """renders the tags and the strings in the content"""
        file_out.write(f'<{tag_name}>{content}</{tag_name}>')

**Comment by natasha-aleksandrova** [comment]
@jayjonesseattle the code you have so far looks to be step 1 and looks good.


is there a particular step you're stuck on?

have you looked through this? https://uwpce-pythoncert.github.io/PythonCertDevel/exercises/html_renderer_tutorial.html

if you still need help, reach out to us via email or post in MS Teams chat channel.

**Comment by jayjonesseattle** [comment]
No, I was not aware of the link you gave me. I will try to use this as a
reference if I can’t figure it out. Thank you!

**Pull request merged**